### ISSUE TITLE
gen_pattern.py doesn't work

### ISSUE THREAD

**Issue by kaixin-bai** [opencv/opencv#21026]
##### Detailed description

I followed [tutorial_camera_calibration_pattern](https://docs.opencv.org/4.x/da/d0d/tutorial_camera_calibration_pattern.html) to generate circle pattern:
```bash
python gen_pattern.py -o circleboard.svg --rows 7 --columns 5 --type circles --square_size 15
```
and then i got error:
```bash
Traceback (most recent call last):
  File "/home/kb/gen_pattern.py", line 217, in <module>
    main()
  File "/home/kb/gen_pattern.py", line 198, in main
    if len(args.markers) % 2 == 1:
TypeError: object of type 'NoneType' has no len()
```
It seems that `-m` is **ONLY** for radon checkerboard, but i cannot generate the other types of pattern without this parameter.

**Comment by alalek** [comment]
relates #20735
/cc @AleksandrPanov

**Comment by AleksandrPanov** [comment]
@alalek fixed by #21028
I've set `force_builders_only=Docs`
Should run CI on other configurations?
